stdlib/Test/src/Test.jl

```diff
@@ -65,15 +65,15 @@ function scrub_exc_stack(stack)
 end
 
 """
-    Result
+    Test.Result
 
 All tests produce a result object. This object may or may not be
 stored, depending on whether the test is part of a test set.
 """
 abstract type Result end
 
 """
-    Pass
+    Test.Pass <: Test.Result
```

Comment:
I added the prefix `Test.` here and to the other types to signify that those types are not exported.  I think `<: Test.Result` also clarifies what they are.